src/Xamarin.Android.Build.Tasks/Tests/Xamarin.Android.Build.Tests/ResolveSdksTaskTests.cs

```diff
@@ -278,5 +289,124 @@ public void ResolveSdkTiming ()
 			Assert.AreEqual (validateJavaVersion.MinimumRequiredJdkVersion, "1.8", "MinimumRequiredJdkVersion should be 1.8");
 			Directory.Delete (Path.Combine (Root, path), recursive: true);
 		}
+
+		[Test]
+		public void TargetFrameworkGreaterThanInstalled ()
+		{
+			var path = Path.Combine ("temp", TestName);
+			var androidSdkPath = CreateFauxAndroidSdkDirectory (Path.Combine (path, "android-sdk"), "26.0.3", new [] {
+				new ApiInfo () { Id = "27", Level = 27, Name = "Oreo", FrameworkVersion = "v8.1", Stable = true },
+			});
+			string javaExe = string.Empty;
+			string javacExe;
+			var javaPath = CreateFauxJavaSdkDirectory (Path.Combine (path, "jdk"), "1.8.0", out javaExe, out javacExe);
+			var referencePath = CreateFauxReferencesDirectory (Path.Combine (path, "references"), new ApiInfo [] {
+				new ApiInfo () { Id = "27", Level = 27, Name = "Oreo", FrameworkVersion = "v8.1", Stable = true },
+				new ApiInfo () { Id = "28", Level = 28, Name = "P",    FrameworkVersion = "v9.0", Stable = true },
+			});
+			IBuildEngine engine = new MockBuildEngine (TestContext.Out);
+			var resolveSdks = new ResolveSdks {
+				BuildEngine = engine,
+				AndroidSdkPath = androidSdkPath,
+				AndroidNdkPath = androidSdkPath,
+				JavaSdkPath = javaPath,
+				ReferenceAssemblyPaths = new [] {
+					Path.Combine (referencePath, "MonoAndroid"),
+				},
+			};
+			var androidTooling = new ResolveAndroidTooling {
+				BuildEngine = engine,
+				AndroidSdkPath = androidSdkPath,
+				AndroidNdkPath = androidSdkPath,
+				UseLatestAndroidPlatformSdk = true,
+			};
+			Assert.IsTrue (resolveSdks.Execute (), "ResolveSdks should succeed!");
+			Assert.IsTrue (androidTooling.Execute (), "ResolveAndroidTooling should succeed!");
+			Assert.AreEqual ("27", androidTooling.AndroidApiLevel, "AndroidApiLevel should be 27");
+			Assert.AreEqual ("v8.1", androidTooling.TargetFrameworkVersion, "TargetFrameworkVersion should be v8.1");
+			Assert.AreEqual ("27", androidTooling.AndroidApiLevelName, "AndroidApiLevelName should be 27");
+			Assert.AreEqual ("28", androidTooling.SupportedApiLevel, "SupportedApiLevel should be 28");
+			Directory.Delete (Path.Combine (Root, path), recursive: true);
+		}
+
+		[Test]
+		public void TargetFrameworkLessThanInstalled ()
```

Comment:
Not sure I get the purpose of this test. The test name is `TargetFrameworkLessThanInstalled`. But the setup in the test seems to indicate that both api 27 and 28 and installed in both directories.. ?